lib/ws-server.js

```diff
@@ -0,0 +1,71 @@
+
+module.exports = WebSocketServer;
+
+
+var assert = require('assert');
+var debug = require('./debug')('process:' + process.pid);
+var message = require('./message');
+var util = require('util');
+var url = require('url');
+var Server = require('ws').Server;
+var WebSocketChannel = require('./ws-channel');
+
+
+function WebSocketServer(options, callback) {
+  if (typeof options === 'string') {
+    options = url.parse(options);
+    options.host = options.hostname;
+  }
+
+  Server.call(this, options, callback);
+
+  this._channels = {};
+  this.on('connection', this._onConnection.bind(this));
+}
+
+util.inherits(WebSocketServer, Server);
+
+
+WebSocketServer.prototype.unref = function unref() {
+  return this._server.unref();
+};
+
+
+WebSocketServer.prototype.createChannel =
+    function createChannel(onRequest, onNotification) {
+  var channel = WebSocketChannel.create(onRequest, onNotification);
+  var token = channel.createToken();
+
+  this._channels[token] = channel;
+
+  return channel;
+};
+
+
+WebSocketServer.prototype._onConnection = function(websocket) {
```

Comment:
Yes, I saw that option existed, but I didn't see what the advantage would be over just dropping unauthenticated clients.